### PR TITLE
Update snowplow analytics - AB#10139

### PR DESCRIPTION
**Pull request by FuriousLlama**
# Fixes or Implements [AB#10139](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10139)

-   [x] Enhancement
-   [ ] Bug
-   [ ] Documentation

## Description

Added snowplow tracker functionality wrapper. Added initial trackers 

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [ ] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [x] Not Required

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/public/snowplow.js [hunk 1]
```diff
@@ -20,7 +20,8 @@
     "https://www2.gov.bc.ca/StaticWebResources/static/sp/sp-2-14-0.js",
     "snowplow"
 );
-var collector = "spt.apps.gov.bc.ca";
+//var collector = "spt.apps.gov.bc.ca";
+var collector = "spm.apps.gov.bc.ca";
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/public/snowplow.js:24]: will update it once I get confirmation from dan

**Review by tiagograf-AOT — APPROVED** [review]

**Review by tiagograf-AOT — CHANGES_REQUESTED** [review]
it seems the tests don't like the change

File: Apps/WebClient/src/ClientApp/public/snowplow.js (hunk 1 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/public/snowplow.js]: working now

**Comment by FuriousLlama** [comment]
> it seems the tests don't like the change

The are fixed now! :D

**Review by tiagograf-AOT — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/test/reports.test.ts [hunk 2]
```diff
@@ -17,13 +17,16 @@ import { RegistrationStatus } from "@/constants/registrationStatus";
...
 import MedicationStatementHistory from "@/models/medicationStatementHistory";
+import { SnowplowWindow } from "@/plugins/extensions";
 import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
 import container from "@/plugins/inversify.container";
 import { ClientModule } from "@/router";
 import { ILogger } from "@/services/interfaces";
 import { GatewayStoreOptions } from "@/store/types";
 import ReportsView from "@/views/reports.vue";
 
+declare let window: SnowplowWindow;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/test/reports.test.ts:28]: Is this doing anything?

**Review by MikeLyttle — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/test/reports.test.ts (hunk 2 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/test/reports.test.ts:28]: I think is needed to get the tests working. But I will double-check once it builds in dev

**Pull request merged**